app/src/main/java/de/westnordost/streetcomplete/quests/max_speed/AddMaxSpeed.kt

```diff
@@ -56,6 +56,16 @@ class AddMaxSpeed : OsmFilterQuestType<MaxSpeedAnswer>() {
                 changes.modify("highway", "living_street")
             }
             is ImplicitMaxSpeed -> {
+                if (answer.countryCode == "GB") {
+                    // nsl_restricted only gets added if the road is lit (whether already set or answered by the user in this quest)
+                    if (answer.roadType == "nsl_restricted") {
+                        changes.add("lit", "yes")
+                    }
+                    // The other answers, nsl_single and nsl_dual, only get added if the road is not lit
+                    else {
+                        changes.add("lit", "no")
+                    }
```

Comment:
but couldn't an `nsl_dual` be lit?

Reply:
Yes, and that's what I'm working on now for the National Speed Limit option, but at the moment `nsl_dual` only gets added if you say the road is not lit:
[AddMaxSpeedForm.kt#216](https://github.com/streetcomplete/StreetComplete/blob/3645254bea414b0b9275b68203fa96ee2fce8f57/app/src/main/java/de/westnordost/streetcomplete/quests/max_speed/AddMaxSpeedForm.kt#L216-L221)